arelle/PluginManager.py

```diff
@@ -22,6 +22,8 @@
     from collections import OrderedDict
 except ImportError:
     OrderedDict = dict # python 3.0 lacks OrderedDict, json file will be in weird order
+from collections.abc import Generator, Callable
```

Comment:
I'd say this is up to you guys and the what guidelines you want us to work under but I don't think I've seen those imports under `if TYPE_CHECKING` before.

Reply:
The import overhead seems minimal (most of these are going to be imported anyway, so it's typically a cheap hash lookup). I'd prefer to limit `TYPE_CHECKING` usage. Use it to avoid circular dependencies and if we actually discover that something introduces overhead.